auth-starter/src/main/java/com/vaadin/auth/starter/VaadinAuthSecurityConfiguration.java

```diff
@@ -199,4 +237,30 @@ CompositeLogoutHandler getLogoutHandler() {
             return logoutHandler;
         }
     }
+
+    static class VaadinExpiredSessionStrategy
+            implements SessionInformationExpiredStrategy {
+
+        private static final Logger LOGGER = LoggerFactory
+                .getLogger(VaadinExpiredSessionStrategy.class);
+
+        @Override
+        public void onExpiredSessionDetected(
+                SessionInformationExpiredEvent event)
+                throws IOException, ServletException {
+            final var request = event.getRequest();
+            final var response = event.getResponse();
+            final var vaadinRequestParameter = request.getParameter("v-r");
+            final var redirectRoute = '/' + request.getContextPath();
+            if ("uidl".equals(vaadinRequestParameter)) {
```

Comment:
nit: Flow has a static enum for it HandlerHelper.RequestType or ApplicationConstants.REQUEST_TYPE_UIDL.

The same goes to the Vaadin-Refresh, even tho I'm not sure if you wanna have a direct dependency to flow-client just for that.

Reply:
Yes, `Vaadin-Refresh` is defined in `flow-client` and we don't want to depend on that. This PR is a draft since it comes from fast prototyping and it still needs some polishing, so string literals and other shortcuts are expected until it's ready for review 🙂